File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/mesh/persistence/androidxml/AndroidXmlLoader.java
package com.abubusoft.xenon.mesh.persistence.androidxml;

import java.io.IOException;
import java.io.InputStream;

import com.abubusoft.xenon.core.XenonRuntimeException;

import com.abubusoft.kripton.KriptonBinder;

import android.content.Context;

public class AndroidXmlLoader {
	public static XmlDataModel parse(InputStream stream) throws IOException, Exception {
		XmlDataModel mesh = KriptonBinder.xmlBind().parse(stream, XmlDataModel.class);
		/*BinderReader binder = BinderFactory.getXMLReader();
		XmlDataModel mesh = binder.read(XmlDataModel.class, stream);*/

		return mesh;
	}

	public static XmlDataModel loadFromAsset(Context context, String fileName) {
		try {
			return parse(context.getAssets().open(fileName));
		} catch (Exception e) {
			e.printStackTrace();
			throw new XenonRuntimeException(e);
		}
	}

	public static XmlDataModel loadFromResources(Context context, int resourceId) {
		try {
			return parse(context.getResources().openRawResource(resourceId));
		} catch (Exception e) {
			e.printStackTrace();
			throw new XenonRuntimeException(e);
		}
	}
}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/mesh/persistence/androidxml/AndroidXmlLoader.kt
package com.abubusoft.xenon.mesh.persistence.androidxml

import android.content.Context
import com.abubusoft.kripton.KriptonBinder
import com.abubusoft.xenon.core.XenonRuntimeException
import java.io.IOException
import java.io.InputStream

object AndroidXmlLoader {
    @Throws(IOException::class, Exception::class)
    fun parse(stream: InputStream?): XmlDataModel {
        /*BinderReader binder = BinderFactory.getXMLReader();
		XmlDataModel mesh = binder.read(XmlDataModel.class, stream);*/return KriptonBinder.xmlBind().parse(stream, XmlDataModel::class.java)
    }

    fun loadFromAsset(context: Context, fileName: String?): XmlDataModel {
        return try {
            parse(context.assets.open(fileName!!))
        } catch (e: Exception) {
            e.printStackTrace()
            throw XenonRuntimeException(e)
        }
    }

    fun loadFromResources(context: Context, resourceId: Int): XmlDataModel {
        return try {
            parse(context.resources.openRawResource(resourceId))
        } catch (e: Exception) {
            e.printStackTrace()
            throw XenonRuntimeException(e)
        }
    }
}